test/scan-test.js

```diff
@@ -1,56 +1,86 @@
 'use strict';
 
-// No `process save-core` on linuxes :(
-if (process.platform !== 'darwin')
-  return;
-
 const tape = require('tape');
-
 const common = require('./common');
 
 tape('v8 findrefs and friends', (t) => {
-  t.timeoutAfter(90000);
+  t.timeoutAfter(common.saveCoreTimeout);
+
+  // Use prepared core and executable to test
+  if (process.env.LLNODE_CORE && process.env.LLNODE_NODE_EXE) {
+    test(process.env.LLNODE_NODE_EXE, process.env.LLNODE_CORE, t);
+  } else if (process.platform === 'linux') {
+    t.skip('No `process save-core` on linux');
+    t.end();
+  } else {
+    saveCoreAndTest(t);
+  }
+});
 
+function saveCoreAndTest(t) {
+  // Create a core and test
   const sess = common.Session.create('inspect-scenario.js');
+  sess.timeoutAfter(common.saveCoreTimeout);
 
   sess.waitBreak(() => {
     sess.send(`process save-core ${common.core}`);
     // Just a separator
     sess.send('version');
   });
 
-  sess.wait(/lldb\-/, () => {
+  sess.wait(/lldb-/, (err) => {
+    t.error(err);
     t.ok(true, 'Saved core');
+    sess.send('target delete 0');
+    sess.quit();
 
-    sess.send(`target create -c ${common.core}`);
+    test(process.execPath, common.core, t);
   });
-
-  sess.wait(/Core file[^\n]+was loaded/, () => {
+}
+
+function test(executable, core, t) {
+  let sess, ranges;
+  if (process.env.LLNODE_NO_RANGES) {
```

Comment:
This is an environment variable for testing linux coredumps on Mac with newer versions of lldb (which does not need ranges file and setting one would result in empty scans). We can also just port the lldb version detection part from `configure` over and skip the ranges generation on higher versions of lldb (also that way we don't even to configure `TEST_LLNODE_BINARY` in the CI/on Linux anymore)